### PR TITLE
[WIP] architecture block.

### PR DESCRIPTION
**Pull request by pashabetenya**

### CODE REVIEW

**Review by qkul — CHANGES_REQUESTED** [review]

File: lib/ui/pages/feed/home/currency_item.dart [hunk 1]
```diff
@@ -1,13 +1,13 @@
 import 'package:flutter/material.dart';
```
[the remark this answers is not in the document]
Comment (pashabetenya) [reply lib/ui/pages/feed/home/currency_item.dart:1]: done.

File: lib/core/bloc/home_page_bloc.dart [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  }
+}
+
+// -- -- -- --
+// STATS.
+// -- -- -- --
+
+abstract class HomeState {}
+
+class HomeStateInit extends HomeState {}
```
[the remark this answers is not in the document]
Comment (pashabetenya) [reply lib/core/bloc/home_page_bloc.dart:25]: done.

File: lib/core/bloc/shell_page_bloc.dart [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+          );
+    }
+  }
+}
+
+// -- -- -- --
+// EVENTS.
+// -- -- -- --
+
+abstract class ShellEvent {
```
[the remark this answers is not in the document]
Comment (pashabetenya) [reply lib/core/bloc/shell_page_bloc.dart:34]: done.

File: lib/core/bloc/home_page_bloc.dart [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      yield GotCurrencyState(_currencyList);
+    }
+  }
+}
+
+// -- -- -- --
+// STATS.
+// -- -- -- --
+
+abstract class HomeState {}
```
[the remark this answers is not in the document]
Comment (pashabetenya) [reply lib/core/bloc/home_page_bloc.dart:23]: done.

File: lib/core/bloc/shell_page_bloc.dart [hunk 5]
```diff
@@ -0,0 +1,55 @@
+import 'package:cryptocurrency/core/services/navigation_service.dart';
```
[the remark this answers is not in the document]
Comment (pashabetenya) [reply lib/core/bloc/shell_page_bloc.dart:1]: ok, i get it. i'll try to figure it out and fix it.

File: lib/core/services/dependency_service.dart [hunk 6]
```diff
@@ -3,22 +3,28 @@ import 'package:cryptocurrency/core/blocs/shell/shell_bloc.dart';
...
-    getIt.registerFactory(() => ShellBloc());
-    getIt.registerFactory(() => HomeBloc());
-  }
+//   static void _registerBlocs(GetIt getIt) {
+//     getIt.registerFactory(() => ShellBloc());
+//     getIt.registerFactory(() => HomeBloc());
+//   }
+// }
+
+GetIt locator = GetIt.instance;
```
Comment (pashabetenya) [inline lib/core/services/dependency_service.dart:26]: to go back to the old logic that was there before.

File: lib/core/services/dependency_service.dart [hunk 7]
```diff
@@ -3,22 +3,28 @@ import 'package:cryptocurrency/core/blocs/shell/shell_bloc.dart';
...
+
+void setupLocator() {
+  locator.registerLazySingleton(() => NavigationService());
```
Comment (pashabetenya) [inline lib/core/services/dependency_service.dart:29]: add registration for HomeBloc, ShellBloc.

File: lib/ui/pages/auth/login_page.dart [hunk 8]
```diff
@@ -118,7 +119,7 @@ class _LoginPageState extends State<LoginPage> {
       child: TextButton(
         onPressed: () {
           // Navigator.of(context).pushReplacementNamed(Routes.forgot);
-          navigationService.navigateTo(Pages.forgot);
+          _navigationService.navigateTo(Pages.forgot);
```
Comment (pashabetenya) [inline lib/ui/pages/auth/login_page.dart:122]: to prescribe the same transitions for other pages.

File: lib/core/services/dependency_service.dart [hunk 9]
```diff
@@ -8,24 +8,14 @@ final _getIt = GetIt.I;
...
-    // return _getIt;
+
+    _registerBlocs(_getIt);
   }
 
-  // static void _registerBlocs(GetIt getIt) {
-  //   getIt..registerFactory(() => ShellBloc());
-  //   getIt..registerFactory(() => HomeBloc());
-  // }
+  static void _registerBlocs(GetIt getIt) {
```
Comment (pashabetenya) [inline lib/core/services/dependency_service.dart:17]: prescribe the architecture (initial) for the authorization pages. start with the login first, and then you will see.

File: lib/main.dart [hunk 10]
```diff
@@ -1,23 +1,15 @@
...
-import 'package:cryptocurrency/core/services/navigation_service.dart';
-// import 'package:cryptocurrency/generated/locale_keys.g.dart';
 import 'package:cryptocurrency/ui/pages/index.dart';
 import 'package:easy_localization/easy_localization.dart';
 import 'package:firebase_core/firebase_core.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter/services.dart';
 import 'package:get/get.dart';
-// import 'package:get/get.dart';
-// import 'package:fluttertoast/fluttertoast.dart';
```
Comment (pashabetenya) [inline lib/main.dart]: where did the error message go? what's the point of the cycle in this case..

Comment (pashabetenya) [reply lib/main.dart]: done.

**Pull request merged**